docs/configuration.md

```diff
@@ -109,15 +109,26 @@ $configuration = Configuration::forAsymmetricSigner(
 
 Currently supported asymmetric algorithms:
 
-| Name    | Description                     | Class                               |
-| ------- | ------------------------------- | ----------------------------------- |
-| `ES256` | ECDSA using P-256 and SHA-256   | `\Lcobucci\JWT\Signer\Ecdsa\Sha256` |
-| `ES384` | ECDSA using P-384 and SHA-384   | `\Lcobucci\JWT\Signer\Ecdsa\Sha384` |
-| `ES512` | ECDSA using P-521 and SHA-512   | `\Lcobucci\JWT\Signer\Ecdsa\Sha512` |
-| `RS256` | RSASSA-PKCS1-v1_5 using SHA-256 | `\Lcobucci\JWT\Signer\Rsa\Sha256`   |
-| `RS384` | RSASSA-PKCS1-v1_5 using SHA-384 | `\Lcobucci\JWT\Signer\Rsa\Sha384`   |
-| `RS512` | RSASSA-PKCS1-v1_5 using SHA-512 | `\Lcobucci\JWT\Signer\Rsa\Sha512`   |
-| `EdDSA` | EdDSA signature algorithms      | `\Lcobucci\JWT\Signer\Eddsa`        |
+| Name    | Description                     | Class                               | Key length req. |
+| ------- | ------------------------------- | ----------------------------------- |-----------------|
+| `ES256` | ECDSA using P-256 and SHA-256   | `\Lcobucci\JWT\Signer\Ecdsa\Sha256` | 224 bits        |
+| `ES384` | ECDSA using P-384 and SHA-384   | `\Lcobucci\JWT\Signer\Ecdsa\Sha384` | 224 bits        |
+| `ES512` | ECDSA using P-521 and SHA-512   | `\Lcobucci\JWT\Signer\Ecdsa\Sha512` | 224 bits        |
+| `RS256` | RSASSA-PKCS1-v1_5 using SHA-256 | `\Lcobucci\JWT\Signer\Rsa\Sha256`   | 2048 bits       |
+| `RS384` | RSASSA-PKCS1-v1_5 using SHA-384 | `\Lcobucci\JWT\Signer\Rsa\Sha384`   | 2048 bits       |
+| `RS512` | RSASSA-PKCS1-v1_5 using SHA-512 | `\Lcobucci\JWT\Signer\Rsa\Sha512`   | 2048 bits       |
+| `EdDSA` | EdDSA signature algorithms      | `\Lcobucci\JWT\Signer\Eddsa`        | 256 bits        |
```

Comment:
Is this correct? AFAIK EdDSA key pairs are 96 bytes long (32 for the public key and 64 for the secret key), so it should be for 512 bits?

Reply:
As far as I know, "key-pair" length isn't a thing, only private key matters in strength and so it is in key length requirement.

And Ed25519 spec within EdDSA uses 32 bytes for private key.
You often see a 64 bytes private key because the public key is appended to the private one (I have no idea why).

Reply:
I'm talking based on these constants in PHP sodium:

```php
const SODIUM_CRYPTO_SIGN_PUBLICKEYBYTES = 32;
const SODIUM_CRYPTO_SIGN_SECRETKEYBYTES = 64;
const SODIUM_CRYPTO_SIGN_KEYPAIRBYTES = 96;
```

> You often see a 64 bytes private key because the public key is appended to the private one (I have no idea why).

Isn't that to represent the pair?

Reply:
https://www.rfc-editor.org/rfc/rfc8032.html#section-5.1.5

> The private key is 32 octets (256 bits, corresponding to b) of cryptographically secure random data.

I've always only used 32 bytes to represent private keys in my production for all `libsodium` related apps, even this one.

:shrug:

Reply:
All good then =)